spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -148,19 +152,23 @@ public String rewrittenFileSetId() {
   }
 
   public DistributionMode distributionMode() {
-    String defaultValue;
-    if (table.sortOrder().isSorted()) {
-      defaultValue = TableProperties.WRITE_DISTRIBUTION_MODE_RANGE;
-    } else {
-      defaultValue = TableProperties.WRITE_DISTRIBUTION_MODE_DEFAULT;
```

Comment:
Do we need to continue to respect this? I think the idea was for this to control what to do for unsorted tables, but now this hard-codes NONE. Should we deprecate the property and remove it in 0.14.0?

Reply:
The problem is that our default value isn't static and cannot be represented by a literal. It is just a constant and we still respect the table property.

I'd probably just deprecate `WRITE_DISTRIBUTION_MODE_DEFAULT`. Do you agree, @rdblue?

Reply:
Sounds fine to me.

Reply:
It turns out this property is used by Flink, which makes more sense as Flink does not support `range`. I'll probably keep things as is then.

Reply:
I think we should follow up and deprecate it. We can do that in a PR that updates Flink to set its own default like here.